### PR TITLE
Use remark presets

### LINKED ISSUE #1843: Use remark presets
Ref: https://github.com/stylelint/stylelint/pull/1826#issuecomment-241642138

### PR DESCRIPTION
**Pull request by jeddy3**
Closes #1843

I figured it'd be easiest if we just went with remark's `consistent` and `recommended` presets. The only things that is at odds with our current style is [`list-item-indent`](https://github.com/wooorm/remark-lint/blob/master/doc/rules.md#list-item-indent), which is set to `"tab-size"` in the preset whereas we use `"space"`. We can either override the preset setting for that rule (which is what currently happens in the PR) or we can update the docs to use the `"tab-size"` setting. I'm easy either way. Although the `"tab-size"` style does look a little funky.

@wooorm A quick question if you have the time? Any advantages to the `"tab-size"` style over the `"space"` one, or is it purely a cosmetic thing?

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@jeddy3 lgtm

**Comment by evilebottnawi** [comment]
@jeddy3 what about create config for us?

**Comment by jeddy3** [comment]
> what about create config for us? 

If our config is nothing more than using the two presets, then it might be more hassle than it's worth. Lets get this in so remark is updated. We can always move to a config later.

**Comment by wooorm** [comment]
> @wooorm A quick question if you have the time? Any advantages to the "tab-size" style over the "space" one, or is it purely a cosmetic thing?

It’s [recommended](https://github.com/wooorm/remark-lint/blob/f5b7bd79b9dde770b3a8abecadfa5defaba5f895/packages/remark-preset-lint-recommended/index.js#L18-L21) because rendering can vary across vendors. So it’s a safe default that ensures your markdown will render the same across gfm, commonmark, and “pedantic” markdown.
That difference only comes into play when nesting deeply though, so you should be mostly fine when using a different style!

File: package.json [hunk 1]
```diff
@@ -118,18 +120,13 @@
     }
   },
   "remarkConfig": {
+    "presets": [
+      "remark-preset-lint-recommended",
+      "remark-preset-lint-consistent"
```
Comment (wooorm) [inline package.json]: @jeddy3 Oh, and if you want you can also drop `remark-preset-` from both lines here.

Comment (jeddy3) [inline package.json]: Roger.

**Comment by jeddy3** [comment]
> It’s recommended because rendering can vary across vendors. So it’s a safe default that ensures your markdown will render the same across gfm, commonmark, and “pedantic” markdown.
That difference only comes into play when nesting deeply though

Thanks @wooorm! This is great to know. We use remark (within phenomic) to process the markdown for the website, and the docs are also rendered on github using gfm. We often use deeply nested lists within the docs. As such, I think it makes sense to go with the recommended setting of `"tab-size"` to give us the best chance of a consistent render.

@stylelint/core I've updated the PR accordingly.

**Comment by davidtheclark** [comment]
Thanks @jeddy3.

**Pull request merged**